src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,245 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String createCarQuery = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement saveCarStatement = connection.prepareStatement(createCarQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            saveCarStatement.setLong(1, car.getManufacturer().getId());
+            saveCarStatement.setString(2, car.getModel());
+            saveCarStatement.executeUpdate();
+            ResultSet resultSet = saveCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create " + car + ". ", throwable);
+        }
+        createCarsAndDrivers(car);
+        return car;
+    }
+
+    private void createCarsAndDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement saveCarDriverStatement =
+                        connection.prepareStatement(query)) {
+            saveCarDriverStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                saveCarDriverStatement.setLong(2, driver.getId());
+                saveCarDriverStatement.executeUpdate();
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create " + car + ". ", throwable);
+        }
+    }
+
+    @Override
+    public Car get(Long id) {
+        String query = "SELECT "
+                + "taxi.cars.id AS car_id, "
```

Comment:
do we need schema name before table name here?

Reply:
removed the maximum number of characters